search/result.cpp

```diff
@@ -72,10 +74,34 @@ std::string Result::GetLocalizedFeatureType() const
   return platform::GetLocalizedTypeName(classif().GetReadableObjectName(m_featureType));
 }
 
+std::string Result::GetLocalizedMainFeatureType() const
+{
+  ASSERT_EQUAL(m_resultType, Type::Feature, ());
+  return platform::GetLocalizedTypeName(classif().GetReadableObjectName(m_mainFeatureType));
+}
+
 std::string Result::GetFeatureDescription() const
 {
   ASSERT_EQUAL(m_resultType, Type::Feature, ());
-  return GetLocalizedFeatureType() + GetDescription();
+  std::string featureDescription;
+  
+  auto const append = [&featureDescription](std::string_view sv)
+  {
+    if (!featureDescription.empty())
+      featureDescription += feature::kFieldsSeparator;
+    featureDescription += sv;
+  };
+  
+  if (m_mainFeatureType != m_featureType && !m_str.empty())
+    append(GetLocalizedMainFeatureType());
+  
+  if (!m_str.empty())
+    append(GetLocalizedFeatureType());
```

Comment:
What if we have an un-named cafe with a toilet?

Reply:
Fixed. Now it will show Cafe as the title and toilet in the description

<img src="https://github.com/organicmaps/organicmaps/assets/47610359/28d402aa-645b-47c8-8cb5-59e7591f2a89" width="320"/>